src/main/java/bio/terra/cli/service/ExternalCredentialsManagerService.java

```diff
@@ -0,0 +1,162 @@
+import bio.terra.cli.service.utils.HttpUtils;
+import bio.terra.externalcreds.api.SshKeyPairApi;
+import bio.terra.externalcreds.client.ApiClient;
+import bio.terra.externalcreds.model.SshKeyPair;
+import bio.terra.externalcreds.model.SshKeyPairType;
+import com.google.auth.oauth2.AccessToken;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.http.util.TextUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+
+public class ExternalCredentialsManagerService {
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceManagerService.class);
+
+  // the Terra environment where the WSM service lives
+  private final Server server;
+
+  // the client object used for talking to WSM
+  private final ApiClient apiClient;
+
+  private ExternalCredentialsManagerService(@Nullable AccessToken accessToken, Server server) {
+    this.server = server;
+    this.apiClient =
+        new ApiClient(new RestTemplate(List.of(new MappingJackson2HttpMessageConverter())));
+
+    this.apiClient.setBasePath(server.getExternalCredsUri());
+    if (accessToken != null) {
+      // fetch the user access token
+      // this method call will attempt to refresh the token if it's already expired
+      this.apiClient.setAccessToken(accessToken.getTokenValue());
+    }
+  }
+
+  public static ExternalCredentialsManagerService returnFromContext() {
+    return new ExternalCredentialsManagerService(
+        Context.requireUser().getUserAccessToken(), Context.getServer());
+  }
+
+  /**
+   * Gets Ssh key pair from ECM. If one does not exist yet, generate one.
+   *
+   * @param keyPairType git server tied to the key (e.g. GitHub, GitLab, Azure).
+   */
+  public SshKeyPair getSshKeyPair(SshKeyPairType keyPairType) {
+    SshKeyPairApi sshKeyPairApi = new SshKeyPairApi(apiClient);
+    try {
+      return HttpUtils.callWithRetries(
+          () -> sshKeyPairApi.getSshKeyPair(keyPairType),
+          ExternalCredentialsManagerService::isRetryable);
+    } catch (HttpStatusCodeException | InterruptedException e) {
+      if (e instanceof HttpStatusCodeException) {
+        var httpStatusException = (HttpStatusCodeException) e;
+        logErrorMessage(httpStatusException);
+        if (httpStatusException.getStatusCode() == HttpStatus.NOT_FOUND) {
+          return callWithRetries(
+              () -> sshKeyPairApi.generateSshKeyPair(Context.requireUser().getEmail(), keyPairType),
+              "failed to regenerate an ssh key");
+        }
+        throw httpStatusException;
+      }
+      throw new SystemException("Failed to get ssh key", e);
+    }
+  }
+
+  /** Regenerate an SshKey for the user. */
+  public SshKeyPair regenerateSshKeyPair(SshKeyPairType keyPairType) {
```

Comment:
If there's no difference between generate and regenerate, then I'd drop the "re".